SDWebImage/Private/SDAsyncBlockOperation.m

```diff
@@ -35,33 +35,55 @@ + (nonnull instancetype)blockOperationWithBlock:(nonnull SDAsyncBlock)block {
 }
 
 - (void)start {
-    if (self.isCancelled) {
-        return;
-    }
-    
-    [self willChangeValueForKey:@"isExecuting"];
-    self.executing = YES;
-    [self didChangeValueForKey:@"isExecuting"];
-    
-    if (self.executionBlock) {
-        self.executionBlock(self);
-    } else {
-        [self complete];
+    @synchronized (self) {
+        if (self.isCancelled) {
+            self.finished = YES;
+            return;
+        }
+        
+        [self willChangeValueForKey:@"isExecuting"];
+        self.executing = YES;
+        [self didChangeValueForKey:@"isExecuting"];
+        
+        if (self.executionBlock) {
+            self.executionBlock(self);
+        } else {
+            self.executing = NO;
```

Comment:
This sounds that we should also notify the value's change of `self.executing = NO;`.

Reply:
@kinarobin  I rewrite this line by using the setter to trigger the KVO observation notiffication. Review again.

The correct NSOperatio nsubclass, see: https://www.avanderlee.com/swift/asynchronous-operations/
Also, see our `SDWebImageDownloadOperation`, the `isConcurrent` (or `isAsynchronous`, same effect) must be override and return YES. (deafult is NO)